File: cucumber/test-sandbox/src/js/bank-account/bank-account.ts
export default class BankAccount {
    private _balance: number;

    constructor() {
        this._balance = 0;
    }

    public deposit(amount: number) {
        this.balance = Number(this.balance) + Number(amount);
    }

    public withdraw(amount: number) {
        if (Number(amount) > this.balance) return;
        this.balance = Number(this.balance) - Number(amount);
    }

    public set balance(amount: number) {
        this._balance = amount;
    }

    public get balance() {
        return this._balance;
    }
}

File: Java-teaching/03-basic-classes/src/io/nology/BankAccount.java
package io.nology;

public class BankAccount {
    private int balance;
    private boolean isOpen;
    // final - immutable, unchangeable
    // static - stored on the class not the instance!!!
    public final static String description = "THE FINAL DESCRIPTION";
    public static double interestRate;

    public BankAccount(int balance) {
        this.balance = balance;
        this.isOpen = true;
    }

    public void deposit(int amount){
        balance += amount;
    }

    // Write a method that withdraws money only if there is enough money in the account. Otherwise, print a message saying NO.
    // Extension: Throw an exception if there is no money
    // Extension++: Throw a custom BankAccountActionInvalidException

    public void withdraw(int amount) {

        if (balance > amount) {
            balance -= amount;
        } else {
            System.out.println("NO");
        }
    }

    public int getBalance() {
        return balance;
    }

    public void setBalance(int balance) {
        this.balance = balance;
    }
}
